src/rdb.c

```diff
@@ -1186,6 +1212,25 @@ int rdbSaveInfoAuxFields(rio *rdb, int rdbflags, rdbSaveInfo *rsi) {
         if (rdbSaveAuxFieldStrInt(rdb, "repl-offset", server.master_repl_offset) == -1) return -1;
     }
     if (rdbSaveAuxFieldStrInt(rdb, "aof-base", aof_base) == -1) return -1;
+
+    /* Handle additional dynamic aux fields */
+    if (rdbAuxFields != NULL) {
+        dictIterator *di = dictGetIterator(rdbAuxFields);
+        dictEntry *de;
+        while ((de = dictNext(di)) != NULL) {
+            rdbAuxFieldCodec *codec = (rdbAuxFieldCodec *)dictGetVal(de);
+            sds s = codec->encoder(rdbflags);
+            if (s == NULL) continue;
+            if (rdbSaveAuxFieldStrStr(rdb, dictGetKey(de), s) == -1) {
+                sdsfree(s);
+                dictReleaseIterator(di);
+                return -1;
+            }
+            sdsfree(s);
+        }
+        dictReleaseIterator(di);
+    }
```

Comment:
```suggestion
    /* Handle additional dynamic aux fields */
    if (rdbAuxFields != NULL) {
        dictIterator di;
        dictInitIterator(&di, rdbAuxFields)
        dictEntry *de;
        while ((de = dictNext(&di)) != NULL) {
            rdbAuxFieldCodec *codec = (rdbAuxFieldCodec *)dictGetVal(de);
            sds s = codec->encoder(rdbflags);
            if (s == NULL) continue;
            if (rdbSaveAuxFieldStrStr(rdb, dictGetKey(de), s) == -1) {
                sdsfree(s);
                return -1;
            }
            sdsfree(s);
        }
    }
```
Minor, but I've been trying to encourage it since it makes cleanup a bit nicer.